### ISSUE TITLE
Batch matrix multplication

### ISSUE THREAD

**Issue by ketranm** [twitter/torch-autograd#38]
I'm using nn.MM for batch matrix multiplication but it I got an error in torch-autograd. Does torch-autograd support 3D Tensor multiplication at the current state?

Here's is the example code I ran

```lua
local t = require 'torch'
local d = require 'autograd'

local mm = d.nn.MM(false, true)
-- toy examples
params = {}
params.x = torch.randn(2,1,5)
params.y = torch.randn(2,3,5)

test = function(params)
    local w = mm({params.x, params.y})
    return t.sum(w)
end

dt = d(test)
dparams, loss = dt(params)
print(loss)
```
And I got the following errors
```
/torch/install/share/lua/5.1/autograd/nnwrapper.lua:110: attempt to call method 'type' (a nil value)
stack traceback:
	...etran/torch/install/share/lua/5.1/autograd/nnwrapper.lua:110: in function 'fun'
	...rch/install/share/lua/5.1/autograd/direct/DirectNode.lua:73: in function 'mm'
	toy.lua:11: in function 'fun'
	...rch/install/share/lua/5.1/autograd/direct/DirectTape.lua:18: in function 'funOnly'
	...rch/install/share/lua/5.1/autograd/direct/DirectTape.lua:100: in function 'dt'
	toy.lua:16: in main chunk
	[C]: in function 'dofile'
	...tran/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x010ec1b2d0
```

**Comment by alexbw** [comment]
We've seen this error in another context, we'll try to patch it up soon.

**Comment by alexbw** [comment]
So, this is a bug. Modules that take in table arguments do not currently work. We'll have a patch soon.

**Comment by ketranm** [comment]
Awesome. Thanks!

**Comment by alexbw** [comment]
@ketranm Fixed in #40. Let me know if it works for you.
